### ISSUE TITLE
Support single (non-extended) keys

### ISSUE THREAD

**Issue by notmandatory** [bitcoindevkit/bdk-ffi#391]
**Describe the enhancement**  

Currently `DescriptorPublicKey`, `DescriptorPrivateKey`, and `Descriptor` only support extended keys but should also be able to use `BdkDescriptorSecretKey::Single` variants.

**Use case**  

Paperwallets and devices like SatsCards use `Single` keys and to "sweep" them into a BDK wallet we'll need to be able to use them.

**Comment by reez** [comment]
@thunderbiscuit 

Just wanted to double check any thoughts you might have before I possibly close this (Or we could keep/rewrite if want to keep something open about better DescriptorSecretKey::Single ergonomics if there's something to improved on there, etc)

I believe current API supports single key descriptor construction: new_pk, new_pkh, new_wpkh, new_sh_wpkh in descriptor.rs , with Android tests in DescriptorTest.kt.

**Comment by thunderbiscuit** [comment]
Yes this is good to close. One thing I noticed while playing with it a bit this morning is that our constructors like new_pk and new_pkh take strings for keys, and (1) they could instead take a typed version of the keys because we have them potentially in `PublicDescriptorKey` and whatnot, and (2) our keys story is a not very flexible at the moment. The API in this library can only create extended keys, and they _must_ come from mnemonics first. So you have to do something like entropy -> mnemonic -> extendedkey, then not extend it, then toString, then use that xpub inside your `Descriptor::new_pkh` constructor. Not super ergonimic.
